### PR TITLE
User can tap Login button on Launch screen

### PR DESCRIPTION
**Pull request by andela-cofor**
#### What does this PR do?
adds login button 

#### Description of Task to be completed?
When I click the login button, i am navigated to the Login screen
Functionality is properly tested. i.e Test coverage for related code > 85%

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### Screenshots 
screen shot 2018-01-27 at 4 28 17 pm

#### Questions:
N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,5 +7,6 @@
 /build
 /captures
 .externalNativeBuild
+/.idea/misc.xml
```
Comment (oofoamalu) [inline .gitignore]: This is a duplicate. You can remove this line

File: app/build.gradle [hunk 2]
```diff
@@ -1,6 +1,9 @@
 apply plugin: 'com.android.application'
 
 android {
+    testOptions {
+        unitTests.returnDefaultValues = true
```
Comment (oofoamalu) [inline app/build.gradle]: Can you enlighten me on what this configuration does?

File: .gitignore (hunk 1 above)
Comment (andela-cofor) [reply .gitignore]: Ok, I have removed it.

File: app/build.gradle (hunk 2 above)
Comment (andela-cofor) [reply app/build.gradle]: I was trying to get my tests to run but it did not, I have taken it out thanks for pointing that out.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Java Gradle CircleCI 2.0 configuration file
```
Comment (oofoamalu) [inline .circleci/config.yml]: Revert this. We shouldn't this update on this PR

File: .gitignore [hunk 4]
```diff
@@ -6,4 +6,3 @@
 /build
 /captures
 .externalNativeBuild
-/.idea/modules.xml
```
Comment (oofoamalu) [inline .gitignore]: I'm not sure why you deleted this.

Comment (andela-cofor) [reply .gitignore]: Duplicate

File: .circleci/config.yml (hunk 3 above)
Comment (andela-cofor) [reply .circleci/config.yml]: I see, I don't have that file locally, it appears only on git hub so I assume circleci updated the PR when we installed it, that's why I told you tests are failing...

File: app/src/androidTest/java/com/blackboxteam/app/blackbox/LaunchScreenTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class LaunchScreenTest {
+
+    @Rule
+    public ActivityTestRule<LaunchScreenActivity> mActivityRule = new ActivityTestRule<>(
+            LaunchScreenActivity.class);
+
+    @Test
+    public void navigateToLogin() {
```
Comment (oofoamalu) [inline app/src/androidTest/java/com/blackboxteam/app/blackbox/LaunchScreenTest.java]: This test is incomplete. The standard format is:
- Perform an action
- Verify an outcome/output

Here, you have only performed an action. You would need to verify the outcome of that action.

File: app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java [hunk 6]
```diff
@@ -14,21 +14,21 @@
 
 public class LaunchScreenActivity extends AppCompatActivity implements View.OnClickListener {
 
-    @BindView(R.id.secureButton) Button nextScreen;
+    @BindView(R.id.launchScreenLoginButton) Button mloginScreen;
```
Comment (oofoamalu) [inline app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java]: As a style guide, the annotation should be on top the variable. Move the button variable to the next line.

File: app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java [hunk 7]
```diff
@@ -14,21 +14,21 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_launch_screen);
         ButterKnife.bind(this);
 
-        nextScreen.setOnClickListener(this);
+        mloginScreen.setOnClickListener(this);
     }
 
     @Override
     public void onClick(View v) {
```
Comment (oofoamalu) [inline app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java]: There is a better way to set click listeners. Remove the activity level click listener declaration.

File: app/src/main/res/values/colors.xml [hunk 8]
```diff
@@ -4,8 +4,11 @@
     <color name="colorPrimaryDark">#6e6e6e</color>
     <color name="colorAccent">#FF4081</color>
 
-    <color name="secureButtonBackgroundColor">#5c5c5c</color>
-    <color name="secureButtonTextColor">#fff</color>
+    <color name="buttonBackgroundColor">#5c5c5c</color>
```
Comment (oofoamalu) [inline app/src/main/res/values/colors.xml]: Renaming this variable caused additional changes to files that are not relevant to this PR. Find a cleaner way to define this colour variable.

File: app/src/main/res/values/colors.xml [hunk 9]
```diff
@@ -4,8 +4,11 @@
...
+    <color name="launchScreenButtonTextColor">#fff</color>
```
Comment (oofoamalu) [inline app/src/main/res/values/colors.xml]: Same for this.

File: app/src/androidTest/java/com/blackboxteam/app/blackbox/LaunchScreenTest.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+import android.support.test.runner.AndroidJUnit4;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static junit.framework.Assert.assertNotNull;
+
+/**
+ * Created by oforchinedu on 1/29/18.
```
Comment (oofoamalu) [inline app/src/androidTest/java/com/blackboxteam/app/blackbox/LaunchScreenTest.java]: Java doc should be descriptive. Update comment to explain what the class does.

**Review by oofoamalu — CHANGES_REQUESTED** [review]
Good work overrall. Please address all PR comments.

**Comment by andela-cofor** [comment]
WIP

**Review by oofoamalu — CHANGES_REQUESTED** [review]
@andela-eshaibu It wasn't necessary to change the gradle file configurations. Undo these changes.

File: app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java [hunk 11]
```diff
@@ -9,29 +9,26 @@
...
 
-    @Override
-    public void onClick(View v) {
-        switch (v.getId()) {
-            case R.id.secureButton:
-                Intent intent = new Intent(LaunchScreenActivity.this, LoginScreenActivity.class);
-                startActivity(intent);
-        }
+    @OnClick(R.id.launchScreenLoginButton)
+    public void launchScreenLoginButton(){
```
Comment (oofoamalu) [inline app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java]: Add a space before the opening brace i.e {

File: gradle/wrapper/gradle-wrapper.properties [hunk 12]
```diff
@@ -1,6 +1,7 @@
-#Sat Feb 03 00:36:23 WAT 2018
+#Tue May 30 12:48:27 WAT 2017
```
Comment (oofoamalu) [inline gradle/wrapper/gradle-wrapper.properties]: Undo changes to this file.

File: build.gradle [hunk 13]
```diff
@@ -7,10 +7,9 @@ buildscript {
             url 'https://maven.google.com/'
```
Comment (oofoamalu) [inline build.gradle:7]: Undo changes to this file. We want to use the most recent build versions

File: app/build.gradle [hunk 14]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
```
Comment (oofoamalu) [inline app/build.gradle]: Undo changes to this file. We want to use the latest versions

File: app/build.gradle [hunk 15]
```diff
@@ -23,15 +23,20 @@ android {
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
-    androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2', {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (oofoamalu) [inline app/build.gradle]: These should be implementation not compile

Comment (oofoamalu) [reply app/build.gradle]: @andela-cofor

**Review by oofoamalu — APPROVED** [review]

**Pull request merged**